src/webui/api/torrentscontroller.cpp

```diff
@@ -559,12 +559,26 @@ void TorrentsController::filePrioAction()
     checkParams({"hash", "id", "priority"});
 
     const QString hash = params()["hash"];
-    int fileID = params()["id"].toInt();
-    int priority = params()["priority"].toInt();
+    const QStringList fileIDs {params()["id"].split('|')};
+    bool ok = false;
+    const int priority = params()["priority"].toInt(&ok);
+    if (!ok)
+        throw APIError(APIErrorType::BadParams, tr("Priority must be an integer"));
+
     BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
+    if (!torrent || !torrent->hasMetadata())
+        throw APIError(APIErrorType::BadParams, tr("Torrent hash is not valid"));
+
+    QList<int> fileIdsInts;
```

Comment:
please keep the code succinct, with this var introduced `QStringList fileIDs` is redundant now.

Reply:
fileIDs